src/com/google/enterprise/adaptor/database/UniqueKey.java

```diff
@@ -160,6 +161,57 @@
     return aclSqlCols;
   }
 
+  /**
+   * Attempt to extract ColumnTypes from the supplied Map
+   * of column names to java.sql.Type
+   *
+   * @param sqlTypes Map of column names to Integer java.sql.Types
+   */
+  void addColumnTypes(Map<String, Integer> sqlTypes) {
```

Comment:
There are other reasons to want the Builder. There are number of methods that are used either only from init, while the unique key is being built, or from the tests, poking at the details of the parsing. We can put all of those on the Builder class and reduce the actual UniqueKey to its post-construction production methods. (We can also add some tests of the production methods using crafted prepared statements.) Also, we are thinking of splitting the db.singleDocContentSqlParameters and db.aclSqlParameters arguments into separate setters on the builder, because they like everything else rely on the parsed db.uniqueKey, which will be the sole builder constructor argument.